control/tests/iosys_test.py

```diff
@@ -802,10 +802,64 @@ def test_named_signals(self):
         )
         ss_feedback = ct.feedback(self.mimo_linsys1, self.mimo_linsys2)
         lin_feedback = ct.linearize(ios_connect, 0, 0)
-        np.testing.assert_array_almost_equal(ss_feedback.A, lin_feedback.A)
-        np.testing.assert_array_almost_equal(ss_feedback.B, lin_feedback.B)
-        np.testing.assert_array_almost_equal(ss_feedback.C, lin_feedback.C)
-        np.testing.assert_array_almost_equal(ss_feedback.D, lin_feedback.D)
+        for M, N in ((ss_feedback.A, lin_feedback.A), (ss_feedback.B, lin_feedback.B),
+                     (ss_feedback.C, lin_feedback.C), (ss_feedback.D, lin_feedback.D)):
+            np.testing.assert_array_almost_equal(M, N)
+
+        # We also enforce generic names to be present when systems are created
+        # without names (sys, states, inputs and outputs)
+        def check_name_and_ios_counts(sys):
+            self.assertIsNotNone(sys.name)
+            self.assertEqual(len(sys.state_index), sys.nstates)
+            self.assertEqual(len(sys.input_index), sys.ninputs)
+            self.assertEqual(len(sys.output_index), sys.noutputs)
```

Comment:
Use docstring for function.

This only checks for the correct count.

It would be worth to also check that desired names are really set or have the correct fallback name (if you choose to implement my suggestion in https://github.com/python-control/python-control/pull/400#discussion_r455335103)

Reply:
Done for system names. Still need to do signal names.